src/Reflection/Annotations/AnnotationsPropertiesClassReflectionExtension.php

```diff
@@ -37,16 +37,25 @@ private function findClassReflectionWithProperty(
 		string $propertyName,
 	): ?PropertyReflection
 	{
-		$propertyTags = $classReflection->getPropertyTags();
-		if (isset($propertyTags[$propertyName])) {
+		$propertyReadTags = $classReflection->getPropertyReadTags();
+		$propertyWriteTags = $classReflection->getPropertyWriteTags();
+		if (isset($propertyReadTags[$propertyName]) || isset($propertyWriteTags[$propertyName])) {
+			$readableType = !isset($propertyReadTags[$propertyName]) ? null : TemplateTypeHelper::resolveTemplateTypes(
+				$propertyReadTags[$propertyName]->getType(),
+				$classReflection->getActiveTemplateTypeMap(),
+			);
+			$writableType = !isset($propertyWriteTags[$propertyName]) ? null : TemplateTypeHelper::resolveTemplateTypes(
+				$propertyWriteTags[$propertyName]->getType(),
+				$classReflection->getActiveTemplateTypeMap(),
+			);
+
 			return new AnnotationPropertyReflection(
 				$declaringClass,
```

Comment:
The limitation that any `@property{,-read,write}` will shadow all magic properties from parent. But that is probably not needed for most use cases.